client/src/pages/createnewcourse/CreateNewCourse.jsx

```diff
@@ -1,38 +1,139 @@
-import React, { useState } from "react";
+import React, { useState } from 'react';
 import Header from "../../components/Header";
 import BasicCourseInformation from "../../components/courseinformation/BasicCourseInformation";
 import BasicLessonInformation from "../../components/courseinformation/BasicLessonInformation";
 import plusIcon from "../../assets/math-plus-icon.svg";
 import "./CreateNewCourse.css";
+import CourseService from '../../services/CourseService.js';
+import videoService from '../../services/VideoService';
+import LessonService from '../../services/LessonService.js';
 
 export default function CreateNewCourse() {
-    const [lessons, setLessons] = useState([<BasicLessonInformation key={0} />]);
+    const [uploading, setUploading] = useState(false);
+
+    const [courseInfo, setCourseInfo] = useState({
+        lessons: [],
+        title: '',
+        description: '',
+        category: '',
+    });
+
+    const [lessons, setLessons] = useState([{ id: 0, title: '', selectedFile: null, videoLink: '' }]);
 
     const addLesson = () => {
-        setLessons([...lessons, <BasicLessonInformation key={lessons.length} />]);
+        const newId = lessons.length;
+        setLessons([...lessons, {
+            id: newId,
+            title: '',
+            selectedFile: null,
+            videoLink: '' }]);
+    };
+
+    const handleLessonInfoChange = (lessonId, lessonInfo) => {
+        setLessons(prevLessons => {
+            const updatedLessons = prevLessons.map(lesson => {
+                if (lesson.id === lessonId) {
+                    return { ...lesson, ...lessonInfo };
+                }
+                return lesson;
+            });
+            return updatedLessons;
+        });
+    };
+
+    const handleFileInputChange = (e, lessonId) => {
+        const file = e.target.files[0];
+        const newLessons = lessons.map(lesson => {
+            if (lesson.id === lessonId) {
+                return {
+                    ...lesson,
+                    selectedFile: file,
+                    videoLink: file ? file.name : ''
+                };
+            }
+            return lesson;
+        });
+        setLessons(newLessons);
+    };
+
+    const handleCreateCourse = async () => {
+
+        setUploading(true);
+
+        try {
+            const uploadPromises = lessons.map(lesson => videoService.uploadVideo(lesson.selectedFile, lesson.title));
```

Comment:
This workflow is a bit convoluted, it's documented in the PRs that added these but going off the API alone isn't obvious. The `videoUrl` part of a course is only relevant for non self-hosted videos. E.g., a youtube video. If it's self-hosted, then we query that video ourself and the `videoUrl` can be ignored.

You need to create a course, then lessons, THEN upload videos.

The videos are queried in a format of `c{courseId}.l{lessonId}.mp4`. This means when you upload a file to S3, the filename needs to be converted into that format. That's also why you must create a course and lessons first - so that you know what to name the videos you're uploading. Note this [loadVideo](https://github.com/nickerick/skill-magnet/blob/d974789ace74e276d4942449f5d4dcebd3398b93/client/src/pages/courseviewer/CourseViewer.jsx#L29C3-L33C5) method on the course viewer screen. I originally had this in the example screen but I removed it when testing something and apparently forgot to change it back.

So basically after you create all the lessons, you need to map through list of lessons and do something like

```
lessons.map((lesson) =>videoService.uploadVideo(`c{createdCourse.id}.l{lesson.id}.mp4`);
```